controllers/metal3.io/host_state_machine.go

```diff
@@ -268,20 +269,31 @@ func (hsm *hostStateMachine) handleExternallyProvisioned(info *reconcileInfo) ac
 	case hsm.Host.NeedsHardwareProfile():
 		hsm.NextState = metal3v1alpha1.StateMatchProfile
 	default:
-		hsm.NextState = metal3v1alpha1.StateReady
+		hsm.NextState = metal3v1alpha1.StatePreparing
 	}
 	return actionComplete{}
 }
 
+func (hsm *hostStateMachine) handlePreparing(info *reconcileInfo) actionResult {
+	actResult := hsm.Reconciler.actionPreparing(hsm.Provisioner, info)
+	if _, complete := actResult.(actionComplete); complete {
+		hsm.NextState = metal3v1alpha1.StateReady
+	}
```

Comment:
Here in case of completion it's required to clean the `errorCount`: `hsm.Host.Status.ErrorCount = 0`. This could happen in case previously the clean failed and then later went fine

Reply:
Good catch! Yes.

Reply:
Note: I switched back my review to approve (as it was originally) since this is the only change I think it was required, my other comments are just suggestions (looks like GitHub collapsed all my comments to the latest review status)

Reply:
Thanks, done.

Reply:
Should be inside the `if ...; complete {` block.